[DOC]: Describe custom metadata files regex with AttributeDriver::setFileRegex() (#11904)
doctrine/doctrine2: docs/en/reference/advanced-configuration.rst

the arguments proxydir, namespace and className when the proxy file could not
be found.

Custom Metadata Files Regex
---------------------------

> When using the ``AttributeDriver`` in combination with domain-oriented folder structures,

Reviewer: ```suggestion
When using the ``AttributeDriver`` in combination with domain-oriented directory structures,
```

> When using the ``AttributeDriver`` in combination with domain-oriented folder structures,

Reviewer: A concrete example such as

```
User/
├── User.php
├── UserFixture.php
└── Actions/
    └── Register/
        ├── RegisterUserCommand.php
        └── RegisterUserTest.php
```

wouldn't be out of place I think.

Author: Added it as example

it's common to have domain-related classes like tests, fixtures, commands within the same
directory as entity classes. These should typically be excluded from metadata mapping.

You can define a regular expression to filter which files the driver will consider as mapping sources:

.. code-block:: php

    <?php
    use Doctrine\ORM\Mapping\Driver\AttributeDriver;

    $driver = new AttributeDriver([__DIR__ . '/src/Domain/User']);
    $driver->setFileRegex('/^(?!.*(Test|Fixture|Command)\.php$).*\.php$/');

> Provided regular expression is going to be matched against each file name during mapping discovery.

Reviewer: ```suggestion
The provided regular expression is going to be matched against each file name during mapping discovery.
```

In this example, files ending with ``Test.php``, ``Fixture.php``, or ``Command.php`` will be excluded.

> This allows to ensure that only actual entity classes are taken into account when loading metadata,

Reviewer: ```suggestion
This ensures that only actual entity classes are taken into account when loading metadata,
```
